index.bs

```diff
@@ -165,45 +363,42 @@ spec: webidl
       A <dfn export>permission query algorithm</dfn>
     </dt>
     <dd>
-      Takes an instance of the <a>permission descriptor type</a>,
-      an instance of the <a>permission storage type</a> that's currently
-      stored for this <a>permission</a>, and a new or existing instance of
-      the <a>permission result type</a>, and updates the <a>permission
-      result type</a> instance with the query result. Used by
-      {{Permissions}}' {{Permissions/query()}}
-      method and the <a href="#PermissionStatus-update">PermissionStatus
-      update steps</a>. If unspecified, this defaults to the <a>boolean
-      permission query algorithm</a>.
+      Takes an instance of the <a>permission descriptor type</a> and a new or
+      existing instance of the <a>permission result type</a>, and updates the
+      <a>permission result type</a> instance with the query result. Used by
+      {{Permissions}}' {{Permissions/query()}} method and the <a
+      href="#PermissionStatus-update">PermissionStatus update steps</a>. If
+      unspecified, this defaults to the <a>boolean permission query
+      algorithm</a>.
     </dd>
     <dt>
       A <dfn export>permission request algorithm</dfn>
     </dt>
     <dd>
-      Takes the previously-stored instance of the <a>permission storage
-      type</a>, an instance of the <a>permission descriptor type</a>,
-      and a newly-created instance of the <a>permission result type</a>.
-      Shows the user any necessary prompt to try to increase permissions,
-      and updates the instances of the <a>permission storage type</a> and
-      <a>permission result type</a> to match. May return a {{Promise}}
-      if the request can fail exceptionally. (Merely being denied
-      permission is not exceptional.) Used by {{Permissions}}'
-      {{Permissions/request()}} method, which handles
-      reading and writing the <a>permission store</a>. If unspecified, this
-      defaults to the <a>boolean permission request algorithm</a>.
+      Takes an instance of the <a>permission descriptor type</a> and a
+      newly-created instance of the <a>permission result type</a>. Uses the
+      algorithms in <a href="#requesting-more-permission"></a> to show the user
+      any necessary prompt to try to increase permissions, and updates the
+      instance <a>permission result type</a> to match. May return a {{Promise}}
```

Comment:
This algorithm is entirely internal, so what it returns isn't exposed to javascript. That said, I can probably have it possibly-throw exceptions instead of possibly-returning Promises.